### PR TITLE
LPD-20532 Match DSM List and Cards mapped fields with visual component elements

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPD-19954 Match DSM List and Cards mapped fields with visual component elements](https://issues.liferay.com/browse/LPD-19954)

### What is the goal of this PR?

We're changing list and cards sections. This is a simple change. The large part of this PR is about extending and fixing tests for visualization modes. 

There are still a lot of flakiness failures in existing tests (see screenshot), but they all look to me like failures are in tests themselves. We can fix them in followups, this PR is large enough as it.

### What does it look like?

image

image

image

### CODE REVIEW

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/view/visualization_modes/VisualizationModesPage.ts [hunk 1]
```diff
@@ -76,6 +76,31 @@ export class VisualizationModesPage {
...
+		});
+
+		await changeAssignmentButton.waitFor();
+		await changeAssignmentButton.click();
+
+		await this.fieldSelectModalContainer
+			.getByPlaceholder('Search')
+			.waitFor();
+	}
+
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-views-web/pages/view/visualization_modes/VisualizationModesPage.ts:103]: @chestofwonders I copied this bit from https://github.com/liferay-frontend/liferay-portal/pull/4055. It is the part that is fixing tests that recently got broken by addition of list clearing.

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/view/visualization_modes/VisualizationModesPage.ts [hunk 2]
```diff
@@ -108,6 +133,6 @@ export class VisualizationModesPage {
 			})
 			.click();
 
-		await this.toastContainer.isHidden();
+		await expect(this.toastContainer).toBeHidden();
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-views-web/pages/view/visualization_modes/VisualizationModesPage.ts:136]: In this case, `not.toBeInViewport()` did not work. It looks like it's not a simple isVisible bad - inViewport good. It depends on the specific case. The real confirmation is only if playwright shows the expected state on an assertion.

**Comment by markocikos** [comment]
ci:test:playwright

**Comment by markocikos** [comment]
ci:test:relevant

**Review by juanjofgliferay — APPROVED** [review]
LGTM :clap:

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
ci:report:2550818137

**Comment by markocikos** [comment]
No unique relevant failures, based on https://github.com/liferay-frontend/liferay-portal/pull/4068#issuecomment-2003011076

No playwright failures on `visualizationModes.spec.ts` in https://github.com/liferay-frontend/liferay-portal/pull/4068#issuecomment-1999850216

**Comment by markocikos** [comment]
ci:test:stable

**Comment by markocikos** [comment]
ci:forward:force

**Comment by markocikos** [comment]
ci:forward:force

**Pull request closed without merging**